lms/services/h_api.py

```diff
@@ -61,96 +61,9 @@ def get_user(self, username):
 
         return HUser(username=username, display_name=user_info["display_name"])
 
-    def create_user(self, h_user):
-        """
-        Create a user in H.
-
-        :arg h_user: the user to be created in h
-        :type h_user: HUser
-        """
-        user_data = {
-            "username": h_user.username,
-            "display_name": h_user.display_name,
-            "authority": self._authority,
-            "identities": [
-                {
-                    "provider": h_user.provider,
-                    "provider_unique_id": h_user.provider_unique_id,
-                }
-            ],
-        }
-
-        self._api_request("POST", "users", data=user_data)
-
-    def update_user(self, h_user):
-        """
-        Update details for a user in H.
-
-        Currently this only updates the display name.
-
-        :param h_user: the updated user details to send to h
-        :type h_user: HUser
-        """
-        self._api_request(
-            "PATCH",
-            f"users/{h_user.username}",
-            data={"display_name": h_user.display_name},
-        )
-
-    def upsert_user(self, h_user):
-        """
-        Create or update a user in H as appropriate.
-
-        This is equivalent to calling `update_user()` then `create_user()` if
-        that fails.
-
-        :param h_user: the updated user value
-        :type h_user: HUser
-        """
-        try:
-            self.update_user(h_user)
-        except HAPINotFoundError:
```

Comment:
The removed code to do with upserting was the only place we caught HAPINotFoundError and it serves no purpose with these gone.